cpp/src/utilities/column_utils.hpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2019 BlazingDB, Inc.
+ *     Copyright 2019 Eyal Rozenberg <eyalroz@blazingdb.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file Small utility functions for working with @ref gdf_column structures,
+ * which are raw C structures without any guarantees and operators defined for
+ * them, and thus require a bit of care when handling. This care and due
+ * diligence tends to be rather repetitive; hence this file.
+ *
+ * @note since the "right" way to work with these structures is pass references,
+ * but the "C" way of doing the same involves pointers, many functions are
+ * implemented for both.
+ */
+#ifndef UTILITIES_COLUMN_UTILS_HPP_
+#define UTILITIES_COLUMN_UTILS_HPP_
+
+#include <utilities/type_dispatcher.hpp>
+
+#include <cudf.h>
+
+#include <algorithm>
+
+namespace cudf {
+
+template<typename T>
+T* get_data(const gdf_column& column) noexcept { return static_cast<T*>(column.data); }
```

Comment:
Avoid ODR violation, and it'd be nice for this to work in device code. 
```suggestion
__host__ __device__
inline T* get_data(const gdf_column& column) noexcept { return static_cast<T*>(column.data); }
```

Reply:
I did indeed forget to inline.

However, these are strictly-host-side utilities; which is why it's an `.hpp` rather than a `.cuh` file. There are generally no `gdf_column`s on the device side for now AFAICR, so let's not go there.